crates/router/src/db/dispute.rs

```diff
@@ -238,75 +239,96 @@ impl DisputeInterface for MockDb {
             .into())
     }
 
-    async fn find_disputes_by_merchant_id(
+    async fn find_disputes_by_constraints(
         &self,
         merchant_id: &common_utils::id_type::MerchantId,
-        dispute_constraints: api_models::disputes::DisputeListConstraints,
+        dispute_constraints: &disputes::DisputeListConstraints,
     ) -> CustomResult<Vec<storage::Dispute>, errors::StorageError> {
         let locked_disputes = self.disputes.lock().await;
-
-        Ok(locked_disputes
+        let limit_usize = dispute_constraints
+            .limit
+            .unwrap_or(u32::MAX)
```

Comment:
It's not good idea to keep it to MAX if value is not present right?

Reply:
Its for MockDb we can change this.